bokeh/core/properties.py

```diff
@@ -1255,6 +1263,23 @@ def from_json(self, json, models=None):
         else:
             raise DeserializationError("%s expected a dict or None, got %s" % (self, json))
 
+class ColumnData(Dict):
+    """Property holding column data in form of a dict. Also applies
+    encoding and decoding to the data.
+    """
```

Comment:
You could enforce that this Dict is from `String` to `ColumnArray` and have another class `ColumnArray` that handles the from_json and serialize_value on the arrays maybe, to avoid isinstance checks